main.go

```diff
@@ -108,227 +101,208 @@ func runTests(machine *client.RobotClient) {
 	}
 
 	// wheeled base tests
-	logger.Info("Starting wheeled base tests...")
 	wb.baseTests(wheeledBase, odometry, wb.minLinVel, wb.minAngVel)
 
 	// sensor base tests
-	logger.Info("Starting sensor controlled base tests...")
 	sb.baseTests(sensorBase, odometry, sb.minLinVel, sb.minAngVel)
 
 	// encoded motor tests
-	logger.Info("Starting encoded motor tests...")
 	runMotorTests(leftMotor, odometry)
 
 	// controlled motor tests
-	logger.Info("Starting controlled motor tests...")
 	runMotorTests(rightMotor, odometry)
 
 	// single encoder tests
-	logger.Info("Starting encoder tests...")
 	runEncoderTests(leftMotor, leftEncoder)
+	runEncoderTests(rightMotor, rightEncoder)
 
 	// power sensor tests
-	logger.Info("Starting power sensor tests...")
 	runPowerSensorTests(powerSensor)
 
 	// movement sensor tests
-	logger.Info("Starting movement sensor tests...")
 	runMovementSensorTests(movementSensor)
 
 	// grid tests
-	logger.Info("Starting grid test with sensor controlled base...")
 	runGridTest(sensorBase, odometry)
 
-	if len(failedTests) == 0 {
-		logger.Info("All tests passed successfully!")
-	} else {
+	if len(failedTests) != 0 {
 		message := "tests failed: " + fmt.Sprint(len(failedTests)) + "/" + fmt.Sprint(totalTests) + "\n"
 		for i := 0; i < len(failedTests); i++ {
 			message += "- " + (failedTests[i] + "\n")
 		}
-		message += "for more info, see rovercanary.log"
 		sendSlackMessage(message)
 	}
 }
 
 func runBaseTests(b base.Base, odometry movementsensor.MovementSensor, minLinVel, minAngVel float64) {
-	logger.Info("testing SetVelocity")
 	// SetVelocity: linear = minLinVel mm/s, angular = 0 deg/sec
 	if err := setVelocityTest(b, odometry, r3.Vector{Y: minLinVel}, r3.Vector{}); err != nil {
-		logger.Errorf("error setting velocity to linear = %v mm/s and anguar = 0 deg/sec, err = %v", minLinVel, err)
-		failedTests = append(failedTests, fmt.Sprintf("%v SetVelocity: linear = %v, angular = %v", b.Name().ShortName(), minLinVel, 0))
+		storeErr := fmt.Errorf("error setting velocity to linear = %v mm/s and anguar = 0 deg/sec, err = %v", minLinVel, err)
```

Comment:
is it worth putting this error wrapping in the setVelocityTest function? that way there is less repeated code

alan showed me a way to use a defer function, so maybe something like this would work as a quick way to add?
```
var returnedErr error
var err error
defer func() { returnedErr = multierr.Combine(returnedErr, err) }()
```
return returnedErr

Reply:
maybe? I don't really think I want it to be a combined error because I would like to print every error on a new line for formatting the slack message and I'm not sure how to do that but maybe it's still possible?

Reply:
and I think that would mess up the way I'm counting the number of failed tests, by having each failure be a separate entry, but again I'm sure there's a way to still make it work with your suggested change

Reply:
sorry i said this poorly. you would put the defer function within each test(i.e. `setVelocityTest` & `moveStraightTest`), so you don't need the storeErr bit for every test, but rather each function already returns with the formatting you wanted

Reply:
okay I think I understand I'll look at adding that

Reply:
just started to change this but in doing so I think I prefer the current setup, it's clearer to me at least to separate the formatting a little bit otherwise it's a lot in one line. I don't feel super strongly about it tho so if you do I can def go through and make that change

Reply:
lets leave it as is then. if we need to add more tests or adjust the log formatting then we can try to make the change